### PR TITLE
POLIO-2048: convert campaign modal to page

### PR DESCRIPTION
**Pull request by quang-le**
What problem is this PR solving? Explain here in one sentence.

Related JIRA tickets : POLIO-2048

## Self proofreading checklist

- [ ] Did I use eslint and ruff formatters?
- [ ] Is my code clear enough and well documented?
- [ ] Are my typescript files well typed?
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests?
- [ ] Documentation has been included (for new feature)

## Doc

Tell us where the doc can be found (docs folder, wiki, in the code...).

## Changes

- Extract modal state and API calls to reusable hooks
- Leave existing modal in Calendar
- Add campaign details page:
    - add url + route 
    - use modal state + API hooks 
    - Add TopBar
    - Adapt styling

- Improve query cache on campaign page's dropdowns
- Convert some components to TS
- Rewrite `useGenUrl` for the calendar because of weird edge case

## How to test

- Go to campaigns
- Try to create and edit: 
   - you should be redirected to a page
   - with the obr name in the title if it exists
   - validation, cancel and save should work without bug
- Go to Calendar:
    - use calendars "arrows" to move acrosse dates: redirection should work without bug
    - use "edit" icon button: modal should open

## Print screen / video

Campaigns page

https://github.com/user-attachments/assets/d05c8664-97c5-41f2-b614-d829be93d0c5

Calendar

https://github.com/user-attachments/assets/1fc6c0cc-d972-4cf7-8de5-71ea75f69583

## Notes

Implements: https://github.com/BLSQ/iaso/pull/2608

The `BaseInfoForm` will be reworked in the next PR on this code

### CODE REVIEW

File: plugins/polio/js/src/domains/Campaigns/MainDialog/CreateEditDialog.tsx
Comment (beygorghor) [inline plugins/polio/js/src/domains/Campaigns/MainDialog/CreateEditDialog.tsx:1]: ❤️

File: plugins/polio/js/src/domains/Campaigns/RiskAssessment/RiskAssessmentForm.tsx [hunk 1]
```diff
@@ -48,7 +49,7 @@ export const RiskAssessmentForm = () => {
         [
             setFieldValue,
             // values.dg_authorized_at,
-            values.risk_assessment_rrt_oprtt_approval_at,
+            values.risk_assessment_rrt_oprrt_approval_at,
```
Comment (beygorghor) [inline plugins/polio/js/src/domains/Campaigns/RiskAssessment/RiskAssessmentForm.tsx:52]: 🤣

**Review by beygorghor — APPROVED** [review]
Really nice job. I'm just wondering if we can improve the base info tab lay-out when not in polio type, those inputs are sooo big 
Screenshot 2025-12-11 at 10 52 42

**Comment by quang-le** [comment]
> Really nice job. I'm just wondering if we can improve the base info tab lay-out when not in polio type, those inputs are sooo big Screenshot 2025-12-11 at 10 52 42

Yeah, I know, but centering them and limiting their size wasn't super convincing either, so I opted for stability of layout when swithing tabs
